File: src/main/java/ru/ifmo/ted/security/XmlUserDetailsStore.java
package ru.ifmo.ted.security;

import lombok.Getter;
import lombok.Setter;
import org.springframework.security.core.userdetails.UserDetails;

import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlRootElement;
import java.util.List;

@Getter
@Setter
@XmlRootElement(name = "users")
@XmlAccessorType(XmlAccessType.FIELD)
class XmlUserDetailsStore {

    @XmlElement(name = "user")
    List<XmlUserDetails> users;

    UserDetails findBy(String username) {
        for (UserDetails user : users) {
            if (user.getUsername().equals(username)) return user;
        }
        return null;
    }
}

File: src/main/kotlin/ru/ifmo/ted/security/XmlUserDetailsStore.kt
package ru.ifmo.ted.security

import jakarta.xml.bind.annotation.XmlAccessType
import jakarta.xml.bind.annotation.XmlAccessorType
import jakarta.xml.bind.annotation.XmlElement
import jakarta.xml.bind.annotation.XmlRootElement
import org.springframework.security.core.userdetails.User
import org.springframework.security.core.userdetails.UserDetails

@XmlRootElement(name = "users")
@XmlAccessorType(XmlAccessType.FIELD)
class XmlUserDetailsStore(@XmlElement val users: MutableCollection<XmlUserDetails>) {
    fun findBy(username: String?): UserDetails? {
        return users.find { it.username == username }
    }
}